l1-contracts/contracts/bridge/BridgedStandardERC20.sol

```diff
@@ -45,22 +74,20 @@ contract L2StandardERC20 is ERC20PermitUpgradeable, IL2StandardToken, ERC1967Upg
 
     /// @notice Initializes a contract token for later use. Expected to be used in the proxy.
     /// @dev Stores the L1 address of the bridge and set `name`/`symbol`/`decimals` getters that L1 token has.
-    /// @param _l1Address Address of the L1 token that can be deposited to mint this L2 token
+    /// @param _originToken Address of the origin token that can be deposited to mint this bridged token
     /// @param _data The additional data that the L1 bridge provide for initialization.
     /// In this case, it is packed `name`/`symbol`/`decimals` of the L1 token.
-    function bridgeInitialize(address _l1Address, bytes calldata _data) external initializer {
-        if (_l1Address == address(0)) {
+    function bridgeInitialize(address _originToken, bytes calldata _data) external initializer returns (uint256) {
+        if (_originToken == address(0)) {
             revert ZeroAddress();
         }
-        l1Address = _l1Address;
+        originToken = _originToken;
 
-        l2Bridge = msg.sender;
+        nativeTokenVault = msg.sender;
 
         // We parse the data exactly as they were created on the L1 bridge
-        (bytes memory nameBytes, bytes memory symbolBytes, bytes memory decimalsBytes) = abi.decode(
-            _data,
-            (bytes, bytes, bytes)
-        );
+        (uint256 chainId, bytes memory nameBytes, bytes memory symbolBytes, bytes memory decimalsBytes) = DataEncoding
```

Comment:
This might be an issue, legacy tokens that are in the process of being bridged will fail.. but it is for L1->L2 so we can call claimFailedDeposit so it is not too bad